src/main.cpp

```diff
@@ -86,7 +85,8 @@ int main() {
   xbot::service::system::initSystem();
   xbot::service::startRemoteLogging();
 
-  if (!Robot::General::IsHardwareSupported()) {
+  robot = GetRobot();
```

Comment:
need a nullptr check in case no robot could be loaded (illegal flash configuration)